ownCloud File Provider UI/DocumentActionViewController.swift

```diff
@@ -20,11 +20,101 @@ import UIKit
 import FileProviderUI
 import ownCloudApp
 import ownCloudAppShared
+import ownCloudSDK
 
 class DocumentActionViewController: FPUIActionExtensionViewController {
 
-	@IBOutlet var label : UILabel!
-	@IBOutlet var button : ThemeButton!
+	private var coreConnectionStatusObservation : NSKeyValueObservation?
+	weak var core: OCCore?
+	var themeNavigationController : ThemeNavigationController?
+
+	enum ActionExtensionType {
+		case undefined, sharing, links
+	}
+
+	deinit {
+		coreConnectionStatusObservation?.invalidate()
+		coreConnectionStatusObservation = nil
+
+		if let bookmark = core?.bookmark {
+			OCCoreManager.shared.returnCore(for: bookmark, completionHandler: nil)
+		}
+	}
+
+	override func prepare(forAction actionIdentifier: String, itemIdentifiers: [NSFileProviderItemIdentifier]) {
+
+		guard let identifier = itemIdentifiers.first else {
+			extensionContext.cancelRequest(withError: NSError(domain: FPUIErrorDomain, code: Int(FPUIExtensionErrorCode.userCancelled.rawValue), userInfo: nil))
+			return
+		}
+
+		OCItem.registerIcons()
+		let collection = Theme.shared.activeCollection
+		self.view.backgroundColor = collection.toolbarColors.backgroundColor
+
+		themeNavigationController = ThemeNavigationController()
+		if let themeNavigationController = themeNavigationController {
+			view.addSubview(themeNavigationController.view)
+			addChild(themeNavigationController)
+		}
+
+		showCancelLabel(with: "Connecting…".localized)
+
+		var actionTypeLabel = ""
+		var actionExtensionType : ActionExtensionType = .undefined
+		if actionIdentifier == "com.owncloud.FileProviderUI.Share" {
+			actionExtensionType = .sharing
+			actionTypeLabel = "Sharing".localized
```

Comment:
The action name, if changed, also needs to be changed here.

Reply:
adopted